runtime/src/accounts_index.rs

```diff
@@ -769,6 +779,26 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
         }
     }
 
+    pub fn insert_new(
```

Comment:
nit: how about renaming `insert_new() -> insert_new_if_missing()`, and `insert_new_entry() -> insert_new_entry_if_missing()`. 

From the name I thought at first this might not be safe because if it was replacing the pubkey entry every time, the ref count would be incorrect, but looks like the `w_account_entry.update()` below should cover that.

Reply:
Yea. It really is still upsert. Functionally the same, it just doesn't do the initial `get_account_write_entry()` that tries for the key with a read lock before taking the write lock. From `perf`, doing the search in the btree for the key seems to be quite expensive, so when inserting a lot of new keys, doing it twice slows it down a lot and there is no downside of taking `write` if it's single threaded.